src/marqo/utils.py

```diff
@@ -66,3 +71,68 @@ def convert_dict_to_url_params(d: dict) -> str:
+            default_values = func.__defaults__
+
+            # Get the parameter names
+            signature = inspect.signature(func)
+            default_parameter_names = [
+                param.name for param in signature.parameters.values() if param.default is not inspect.Parameter.empty
+            ]
+
+            # Create a dictionary mapping parameter names to their default values
+            defaults_dict = dict(zip(default_parameter_names, default_values))
+
+            # Check if any of the deprecated parameters have values different from their defaults
+            deprecated_params_with_non_default_values = [
+                param for param in deprecated_params if kwargs.get(param) != defaults_dict.get(param)
+            ]
+
+            if deprecated_params_with_non_default_values:
+                deprecated_params_string = ', '.join(deprecated_params_with_non_default_values)
+                warning_msg = \
+                    f"The parameter(s) {deprecated_params_string} are deprecated. " \
+                    f"Please refer to the documentation " \
+                    f"https://docs.marqo.ai/1.3.0/Using-Marqo-Cloud/indexes/#create-index " \
+                    f"for updated parameters names. These parameters will be removed in Marqo 2.0.0."
+
+                # Issue a deprecation warning
+                mq_logger.warn(warning_msg)
+
+            return func(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+def is_key_arguments_default(current_frame, func, arguments_to_ignore=None):
+    """ Checks if all key arguments of the function are default values.
+    Ignores arguments specified in arguments_to_ignore list."""
+    passed_and_default = zip(current_frame.f_locals.items(), inspect.signature(func).parameters.values())
+    for passed_parameter, default_parameter in passed_and_default:
+        if default_parameter.default is not inspect.Parameter.empty:
+            if arguments_to_ignore and passed_parameter[0] in arguments_to_ignore:
+                continue
+            if passed_parameter[1] != default_parameter.default:
+                return False
+    return True
+
+
+def use_one_of_two_arguments(func, value_name, value, deprecated_value, deprecated_value_name):
+    """ Checks if both value and deprecated_value are passed to the function and returns only one of them,
+    prioritising non-deprecated one if so. Raises error if both of them are specified"""
+    default_parameters = inspect.signature(func).parameters
+    value_is_specified = value != default_parameters[value_name].default
+    deprecated_value_is_specified = deprecated_value != default_parameters[deprecated_value_name].default
+    if value_is_specified and deprecated_value_is_specified:
+        raise ValueError(f"{value_name} and {deprecated_value_name} can't be specified at the same time "
+                         f"because they refer to the same value, please use only one at a time")
```

Comment:
We should encourage best usage. If they specify both, tell users to use the new parameter. For example:
```
Parameters `inference_types` and deprecated `inference_node_types` can't be specified at the same time because they refer to the same value. Please only use the `inference_types` parameter. 
```